server/samples/clients/node/appencryption_client.js

```diff
@@ -0,0 +1,284 @@
+        logger.info('oh no... something went terribly wrong!');
+    } else {
+        logger.info(`test completed successfully\n`);
+    }
+}
+
+/**
+ * Executes run_client once
+ * @param client
+ * @returns {Promise<void>}
+ */
+async function run_once(client) {
+    await run_client(client);
+
+    client.close();
+}
+
+/**
+ * Executes run_client until the process is interrupted.
+ * @param client
+ * @returns {Promise<void>}
+ */
+
+async function run_continuously(client) {
+    while (true) {
+        process.on('SIGINT', () => {
+            logger.info('received SIGINT. terminating client');
+            process.exit(0);
+        });
+        process.on('SIGTERM', () => {
+            logger.info('received SIGTERM. terminating client');
+            process.exit(0);
+        });
+        await run_client(client);
+
+        //  Sleep for 1 second before sending the next request
+        sleep.sleep(1);
+    }
+}
+
+/**
+ * Gets a session and calls run_once or run_continuously based on the value of the continuous parameter
+ * @param continuous
+ * @param id
+ * @param callback
+ * @returns {Promise<void>}
+ */
+async function run(continuous, id, appEncryptionProto, callback) {
+    const client = new SessionClient(appEncryptionProto, callback);
+
+    await client.getSession(`partition-${id}`);
+
+    if (continuous) {
+        await run_continuously(client);
+    } else {
+        await run_once(client);
+    }
+}
+
+function loadProtoDef(protoPath) {
+    let appEncryptionDef = protoLoader.loadSync(__dirname + protoPath, {
+        keepCase: true,
+        defaults: true,
+        oneofs: true,
+    });
+
+    let appEncryptionProto = grpc.loadPackageDefinition(appEncryptionDef);
+
+    return appEncryptionProto;
+}
+
+function main() {
+    let requests = [];
+    let argv = yargs.usage('Usage: $0 --socket [string] --continuous [boolean] ' +
+        '--num-clients [num] --proto-path [string]')
+        .options({
+            's': {
+                alias: 'socket',
+                describe: 'The unix domain socket the server is listening on.',
+                type: 'string',
+                demand: false,
+                default: '/tmp/appencryption.sock',
+            },
+            'c': {
+                alias: 'continuous',
+                describe: 'When present the client will run a continuous test.',
+                type: 'boolean',
+                demand: false,
+                default: false,
+            },
+            'n': {
+                alias: 'num-clients',
+                describe: 'The total number of clients to run asynchronously',
+                type: 'num',
+                demand: false,
+                default: 1,
+            },
+            'p': {
+                alias: 'proto-path',
+                describe: 'The path to the proto file for the service',
+                type: 'string',
+                demand: false,
+                default: '../../../../protos/appencryption.proto',
```

Comment:
Additionally, `path.resolve()` can be used here to clean up the path is presented in the help output.
```suggestion
                default: path.resolve(__dirname + '../../../../protos/appencryption.proto'),
```